File: searching and sorting algo/Missing Number in AP.cpp
class Solution{   
public:
    int findMissing(int arr[], int n) {
        int diff= (arr[n-1]-arr[0])/n ;
        for(int i=1;i<n;i++){
            if(arr[i]!=(arr[i-1]+diff))
            return (arr[i-1]+diff);
        }
        return -1;
    }
};

File: searchingAndSorting/Missing_Number_in_AP.java
class Solution {
    int findMissing(int[] arr, int n) {
        // code here
        Arrays.sort(arr);
        int d=(arr[n-1]-arr[0])/n;
        int sum=arr[0];
        for(int i=1;i<n;i++){
            sum=sum+d;
            if(arr[i]!=sum){
                break;
            }
        }
        return sum;
    }
}
